### PR TITLE
Sketching out stores layout

### PR DESCRIPTION
**Pull request by tomikazi**

### CODE REVIEW

File: pkg/store/entities.go [hunk 1]
```diff
@@ -2,54 +2,123 @@
...
-	// ID returns the ID of the device whose control entities it persists.
-	ID() topo.ID
+/*
+Notes:
 
-	// Read accepts a query in form of a list of partially populated entities and returns any
-	// matching entities on the specified channel
-	Read(ctx context.Context, query []*p4api.Entity, ch chan<- p4api.Entity) error
+Create a map per device, per:
+- each table (map[tableId]*table; map[hash(entry.matches)]*entry)
```
Comment (ccascone) [inline pkg/store/entities.go:15]: Probably you already know this, in P4RT entries within a table are uniquely identified by the set of match fields **and** priority

File: pkg/store/entities.go [hunk 2]
```diff
@@ -2,54 +2,123 @@
...
+- all meters (map[meterId]*entry; same for all below)
```
Comment (ccascone) [inline pkg/store/entities.go:16]: Indirect meter entries are identified by their index, so you probably need a nested map with index as a key pointing to the entry

File: pkg/store/entities.go (hunk 1 above)
Comment (tomikazi) [reply pkg/store/entities.go:15]: Yes.

File: pkg/store/entities.go (hunk 2 above)
Comment (tomikazi) [reply pkg/store/entities.go:16]: Yes. I've not yet decided whether to store each cell separately or as one map entry. Eitherway, yes, Index will be required to resolve the meter (and counters) to their individual cells.

Comment (tomikazi) [reply pkg/store/entities.go:16]: My plan is to implement table entries and counters first... then fan out and do the meters and the rest.

**Pull request merged**